app/src/ui/clone-repository/clone-repository.tsx

```diff
@@ -580,9 +585,15 @@ export class CloneRepository extends React.Component<
     const tabState = this.getSelectedTabState()
     const lastParsedIdentifier = tabState.lastParsedIdentifier
 
+    // If there is no path yet, just update the url
+    if (tabState.path === null) {
+      this.setSelectedTabState({ url }, this.validatePath)
+      return
```

Comment:
It's important not to set the `lastParsedIdentifier: parsed` in the tab state before we have the path. Otherwise, next time the path is changed, the app will assume the last path component is the repo name and can be changed when the URL is updated.

For example, if the initial path is `/Users/sergiou87/Documents/GitHub` and the `lastParsedIdentifier` is not `null`, when I change the URL to point to a different repository, it will remove that `GitHub` and put the repo name in there instead of appending it.